identity/aziot-identityd/src/lib.rs

```diff
@@ -28,6 +28,27 @@ const DPS_ASSIGNMENT_RETRY_INTERVAL_SECS: u64 = 10;
 /// This is the number of seconds to wait for DPS to complete assignment to a hub
 const DPS_ASSIGNMENT_TIMEOUT_SECS: u64 = 120;
 
+/// URI query parameter that identifies module identity type.
+const ID_TYPE_AZIOT: &str = "aziot";
+
+/// URI query parameter that identifies local identity type.
+const ID_TYPE_LOCAL: &str = "local";
+
+macro_rules! match_id_type {
+    ($id_type:ident, $(($type:ident, $action:expr)),+) => {
+        if let Some(id_type) = $id_type {
+            match id_type.as_str() {
```

Comment:
If you're just going to use it as a `&str`, then why not take `Option<&str>` instead of `Option<String>` ?

(clippy would've warned you about this but probably didn't because you're doing it inside a macro.)

Reply:
Fixed